v2/emailpassword/troubleshooting/cors-issues.mdx

```diff
@@ -0,0 +1,261 @@
+  cors({
+    origin: "http://127.0.0.1:8080",
+    allowedHeaders: ["content-type", ...SuperTokens.getAllCORSHeaders()],
+    methods: ["GET", "PUT", "POST", "DELETE"],
+    //highlight-start
+    credentials: true, // Check this line
+    //highlight-end
+
+  })  
+);
+```
+
+:::danger
+[Error] Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204
+:::
+
+This happens when the CORS origin field is incorrect. Another varient of this error is when the error code is a 404. Generally, that also means a misconfiguration in the CORS origin field. 
+
+We can remedy this issue by ensuring the origin field lines up with the API domain located in other parts of the code. I’ve highlighted them in bold below -
+
+```tsx
+// api-server.js
+
+import supertokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+import EmailPassword from "supertokens-node/recipe/emailpassword";
+import Session from "supertokens-node/recipe/session";
+import Dashboard from "supertokens-node/recipe/dashboard";
+
+supertokens.init({
+  framework: "express",
+  supertokens: {
+    connectionURI: "https://try.supertokens.com",
+  },
+
+  appInfo: {
+    appName: "SuperTokens Demo App",
+    //highlight-start
+    apiDomain: "http://localhost:8080", // Check this line**
+    //highlight-end
+    websiteDomain: "http://localhost:3000",
+  },
+   recipeList: [
+      EmailPassword.init(),
+      Session.init(),
+      Dashboard.init({
+        apiKey: "supertokens_is_awesome",
+      }),
+   ],
+});
+
+const app = express();
+app.use(
+   cors({
+      //highlight-start
+      origin: "http://localhost:8080", // Check this line
+      //highlight-end
+      allowedHeaders: ["content-type", ...supertokens.getAllCORSHeaders()],
+      methods: ["GET", "PUT", "POST", "DELETE"],
+      credentials: true,
+   })
+);
+```
+
+```tsx
+//App.js
+import SuperTokens from "supertokens-auth-react";
+import EmailPassword from "supertokens-auth-react/recipe/emailpassword";
+import Session from "supertokens-auth-react/recipe/session";
+
+SuperTokens.init({
+ appInfo: {
+    appName: "SuperTokens Demo App",
+    //highlight-start
+    apiDomain: "http://localhost:8080", // Check this line
+    //highlight-end
+    websiteDomain: "http://localhost:3000",
+ },
+ recipeList: [
+    EmailPassword.init(),
+    Session.init()
+ ]
+});
+```
+
+:::danger
+[Error] Origin http://localhost:3000 is not allowed by Access-Control-Allow-Origin. Status code: 200
+:::
+
+This happens when the Supertokens middleware is before the CORS middleware. It can also cause an issue in the context of nodejs app where OPTIONS APIs have the right headers, but when you do a POST / GET request to one of the supertokens APIs, it doesn't have the right headers. This is because the CORS middleware doesn't get called if it's placed after the Supertoeksn middleware. To fix, simply put the CORS middleware before the Supertokens middleware.
```

Comment:
There are two reasons this would happen:
- If the Access-Control-Allow-Origin doesn't have http://localhost:3000 at all in the values, OR
- It has http://localhost:3000, but the CORS middleware is after the supertokens middleware - in this case, the OPTIONS API will have the right value for Access-Control-Allow-Origin, but the actual GET / POST / PUT API won't since the supertokens middelware will return a response before the CORS middleware is even called.

Reply:
Added in!